common/djangoapps/third_party_auth/provider.py

```diff
@@ -161,13 +162,76 @@ class FacebookOauth2(BaseProvider):
         'SOCIAL_AUTH_FACEBOOK_SECRET': None,
     }
 
+
+class SAMLProviderMixin(object):
+    """ Base class for SAML/Shibboleth providers """
+    BACKEND_CLASS = SAMLAuthBackend
+    ICON_CLASS = 'fa-university'
+
     @classmethod
-    def get_email(cls, provider_details):
-        return provider_details.get('email')
+    def get_url_params(cls):
+        """ Get a dict of GET parameters to append to login links for this provider """
+        return {'idp': cls.IDP["id"]}
 
     @classmethod
-    def get_name(cls, provider_details):
-        return provider_details.get('fullname')
+    def is_active_for_pipeline(cls, pipeline):
+        """ Is this provider being used for the specified pipeline? """
+        if cls.BACKEND_CLASS.name == pipeline['backend']:
+            idp_name = pipeline['kwargs']['response']['idp_name']
+            return cls.IDP["id"] == idp_name
```

Comment:
if `cls.BACKEND_CLASS.name != pipeline['backend']`, should this return `False`, or `None` (as is implicitly being done now?)